core/use-swr.ts

```diff
@@ -174,9 +171,17 @@ export const useSWRHandler = <Data = any, Error = any>(
     return [
       () => {
         const newSnapshot = getSelectedCache(getCache())
-        return isEqual(newSnapshot, memorizedSnapshot)
-          ? memorizedSnapshot
-          : (memorizedSnapshot = newSnapshot)
+        const compareResult = isEqual(newSnapshot, memorizedSnapshot)
+        if (compareResult) {
+          memorizedSnapshot.data = newSnapshot.data
```

Comment:
why we update properties when they're equal? maybe also add a comment here to explain?

Reply:
Because the compare process only considers selected fields (the ones that you've read in the getters before). However we want to always copy over and override other fields here too, which is the key of this fix.

Agreed that it's worth a comment, maybe linking to this PR.